src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.State.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.SyncNamespace
+{
+    internal abstract partial class AbstractSyncNamespaceService<TNamespaceDeclarationSyntax, TCompilationUnitSyntax, TMemberDeclarationSyntax>
+        where TNamespaceDeclarationSyntax : SyntaxNode
+        where TCompilationUnitSyntax : SyntaxNode
+        where TMemberDeclarationSyntax : SyntaxNode
+    {
+        internal sealed class State
+        {                                                                        
+            private static readonly SymbolDisplayFormat s_qualifiedNameOnlyFormat = 
+                new SymbolDisplayFormat(globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.Omitted,
+                                        typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces);
+
+            public Solution Solution { get; }
+
+            /// <summary>
+            /// The document in which the refactoring is triggered.
+            /// </summary>
+            public DocumentId OriginalDocumentId { get; }
+
+            /// <summary>
+            /// The refactoring is also enabled for document in a multi-targeting project, 
+            /// which is the only form of linked document allowed. This property returns IDs
+            /// of the original document that triggered the refactoring plus every such linked 
+            /// documents.
+            /// </summary>
+            public ImmutableArray<DocumentId> DocumentIds { get;  }
+
+            /// <summary>
+            /// This is the default namespace defined in the project file.
+            /// </summary>
+            public string DefaultNamespace { get; }
+
+            /// <summary>
+            /// This is the name of the namespace declaration that trigger the refactoring.
+            /// </summary>
+            public string DeclaredNamespace { get; }
+                                                                                                       
+            /// <summary>
+            /// This is the new name we want to change the namespace to.
+            /// Empty string means global namespace, whereas null means change namespace action is not available.
+            /// </summary>
+            public string TargetNamespace { get; }
```

Comment:
I see these (namespaces) getting splited, it is ever actually used as it is (string)?

Reply:
There's a few places use the string form, i.e. creating syntax node for import, text in the code action title, etc.